src/runtime_src/core/tools/xbutil2/OO_P2P.cpp

```diff
@@ -303,12 +316,13 @@ OO_P2P::execute(const SubCmdOptions& _options) const
 
   if (m_help) {
     printHelp();
-    return;
+    throw xrt_core::error(std::errc::operation_canceled);
   }
 
   // Validate the correct action value is used
+  action_type action;
   try {
-     string2action(m_action);
+    action = string2action(m_action);
```

Comment:
After this operation, we know that the action value is good.  No need to check it again later in the code.